test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -1940,13 +1940,13 @@ public async Task RestoreCommand_RestoreExactVersionWithFailingSourceAsync()
                     LockFilePath = Path.Combine(projectDir, "project.lock.json")
                 };
 
-                var command = new RestoreCommand(request);
-
                 // Act
+                var command = new RestoreCommand(request);
                 var result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
 
                 // Assert
-                Assert.True(result.Success);
+                Assert.False(result.Success);
```

Comment:
This is proving to be problematic. 
perhaps this scenario shouldn't be a restorecommand test anymore...
even by doing these 3 lines of code, it still doesn't fail yet.

@nkolev92 - recommendations? (this appears to work well with nuget.exe...but this still appears true (successful) with only these api calls. the exceptions don't even happen during this test now until after this assert.

Reply:
i think my real question is this... i think i need to move this test to the RestoreRunnerTests -- and start using RestoreRunner... sound good.